ext/enchant/enchant.c

```diff
@@ -264,20 +294,24 @@ PHP_FUNCTION(enchant_broker_init)
 {
 	enchant_broker *broker;
 	EnchantBroker *pbroker;
+	zval *this = getThis();
 
 	if (zend_parse_parameters_none() == FAILURE) {
 		RETURN_THROWS();
 	}
 
 	pbroker = enchant_broker_init();
-
 	if (pbroker) {
-		broker = (enchant_broker *) emalloc(sizeof(enchant_broker));
+		if (this) {
+			broker = Z_ENCHANT_BROKER_P(this);
+		} else {
+			object_init_ex(return_value, enchant_broker_ce);
+			broker = Z_ENCHANT_BROKER_P(return_value);
+		}
 		broker->pbroker = pbroker;
-		broker->dict = NULL;
-		broker->dictcnt = 0;
-		broker->rsrc = zend_register_resource(broker, le_enchant_broker);
-		RETURN_RES(broker->rsrc);
+		broker->nb_dict = 0;
+	} else if (this) {
+		zend_throw_exception(spl_ce_RuntimeException, "Can't create a new EnchantBroker", 0);
```

Comment:
@kocsismate @Girgias @cmb69 What's our stance on use of SPL exceptions?

Reply:
I don't really like them, but it's okay for me to use RuntimeException here.

Reply:
The same for me. + In most of the similar cases we simply use `Exception` if I remember correctly.

Reply:
We didn't really decide on it, as the conversion from SPL exceptions to engine Error in SPL is still open.
But in this specific case seems reasonable to me.

Reply:
so, looks like we have concensus to keep this RuntimeException